### PR TITLE
Drag force functions and tests for them

### LINKED ISSUE #224: Drag Model Development: Linear and Quadratic

### PR DESCRIPTION
**Pull request by btrettel**
Fixes #224.

### CODE REVIEW

File: ablateLibrary/particles/drag/quadratic.cpp [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    }
+
+    dragForcePrefactor *= ablate::utilities::MathUtilities::MagVector(dim, relVel);
+
+    for (int n = 0; n < dim; n++) {
+        dragForce[n] = dragForcePrefactor * relVel[n];
+    }
+};
+
+// TODO: tests
```
Comment (jikembo) [inline ablateLibrary/particles/drag/quadratic.cpp:20]: remove unnecessary comments. Also add a reference on the drag force model and write down the quadratic drag formula in the comments.

**Review by jikembo — CHANGES_REQUESTED** [review]

File: ablateLibrary/particles/drag/quadratic.cpp (hunk 1 above)
Comment (btrettel) [reply ablateLibrary/particles/drag/quadratic.cpp:20]: I will fix this shortly. I forgot to check my to-do list for the formulas and references.

File: ablateLibrary/particles/inertial.hpp [hunk 2]
```diff
@@ -1,5 +1,6 @@
 #ifndef ABLATELIBRARY_INERTIAL_HPP
 #define ABLATELIBRARY_INERTIAL_HPP
+#include "drag/dragModel.hpp"
```
Comment (mmcgurn) [inline ablateLibrary/particles/inertial.hpp:3]: I don't think that the dragModel is being used yet in inertial, I would remove it for now.

File: tests/ablateLibrary/particles/drag/dragModelTests.cpp [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+    PetscReal muF;
+    PetscReal rhoF;
+    PetscReal partDiam;
+    std::vector<PetscReal> expectedDragForce;
+};
+
+class DragModelTestFixture : public testingResources::PetscTestFixture, public ::testing::WithParamInterface<DragModelTestParameters> {};
+
+TEST_P(DragModelTestFixture, ShouldComputeCorrectDragForce) {
+    // arrange
```
Comment (mmcgurn) [inline tests/ablateLibrary/particles/drag/dragModelTests.cpp:22]: Can we add some 1D tests as well?

File: ablateLibrary/particles/drag/linear.cpp [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+     * References: E. Loth, “Quasi-steady shape and drag of deformable bubbles and drops,” International Journal of Multiphase Flow, vol. 34, no. 6, pp. 523–546, Jun. 2008,
+     * doi: 10.1016/j.ijmultiphaseflow.2007.08.010. (See eq. 3.)
+     */
+
+    PetscReal dragForcePrefactor = -3.0 * PETSC_PI * partDiam * muF;
+
+    for (int n = 0; n < dim; n++) {
+        dragForce[n] = dragForcePrefactor * (partVel[n] - flowVel[n]);
+    }
+};
```
Comment (mmcgurn) [inline ablateLibrary/particles/drag/linear.cpp:20]: Can you register this classes with the the parser (https://github.com/UBCHREST/CppParser).  I would use other classes as references?  We can't test it until we get an integration or use a test function with mocks like (https://github.com/UBCHREST/ablate/blob/4c6bd7f7d9856b6e0070385051509f9353cbd23f/tests/ablateLibrary/mathFunctions/simpleFormulaTests.cpp#L11).

**Review by mmcgurn — CHANGES_REQUESTED** [review]
Thank you for the PR!  Normally I would like to see an integration level test but because we are splitting this into the sub-models and then integration with particles as two pr it can wait.  

Would you mind adding some documentation for the drag models in the docs folder so that it will appear on ablate.dev.  Take a look at https://github.com/UBCHREST/ablate#documentation  for more details or let me know if you have any questions.  Thanks!

File: tests/ablateLibrary/particles/drag/dragModelTests.cpp (hunk 3 above)
Comment (btrettel) [reply tests/ablateLibrary/particles/drag/dragModelTests.cpp:22]: I added 1D tests.

File: ablateLibrary/particles/inertial.hpp (hunk 2 above)
Comment (btrettel) [reply ablateLibrary/particles/inertial.hpp]: I removed this for now, will add later as appropriate.

File: ablateLibrary/particles/drag/linear.cpp (hunk 4 above)
Comment (btrettel) [reply ablateLibrary/particles/drag/linear.cpp:20]: I will talk to you about this tomorrow.

File: docs/content/particleFormulations/inertialParticles/GoverningEquations.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+$$\begin{eqnarray} \frac{d\boldsymbol{x}_p}{dt} = \boldsymbol{u}_p \\ \frac{d\boldsymbol{u}_p}{dt} =
+\frac{f}{\tau_p}\left(\boldsymbol{u}_f - \boldsymbol{u}_p\right) + \boldsymbol{g}\left(1-\frac{\rho_f}{\rho_p}\right)
+\end{eqnarray}$$
+
+where, $$f=1+0.15Re_p^{0.687}$$ is the Schiller-Naumann correction factor to Stokes drag expression (see the reference) to account for the effect of finite particle Reynolds number (denoted by $$Re_p=\rho_f|\boldsymbol{u}_f-\boldsymbol{u}_p|d_p/\mu$$). In addition, $$\tau_p=\rho_pd^2_p/18\mu$$ is the particle relaxation time obtained based on Stokes drag expression. $$\mu$$ and $$\rho_f$$ are the carrier phase dynamic viscosity and density, respectively. The upstream fluid velocity for each particle, $$\boldsymbol{u}_f$$, is obtained by interpolating the surrounding fluid velocity to the particle location.
```
Comment (mmcgurn) [inline docs/content/particleFormulations/inertialParticles/GoverningEquations.md:16]: @btrettel It looks like there may be a formatting issue with this page.  Would you mind taking a look at it?

image

Comment (btrettel) [reply docs/content/particleFormulations/inertialParticles/GoverningEquations.md:16]: Seems to be the result of me trying CLion's auto-formatting on Markdown. I'll revert the page to the previous version.

**Review by mmcgurn — APPROVED** [review]

**Review by jikembo — APPROVED** [review]

**Pull request merged**